agent/sidecar_service.go

```diff
@@ -202,14 +202,16 @@ func (a *Agent) sidecarPortFromServiceID(sidecarCompoundServiceID structs.Servic
 	return sidecarPort, nil
 }
 
-func sidecarDefaultChecks(serviceID string, localServiceAddress string, port int) []*structs.CheckType {
+func sidecarDefaultChecks(serviceID string, proxyAddress string, port int) []*structs.CheckType {
+	if proxyAddress == "" {
```

Comment:
`sidecar.Address` is passed in here. If empty, it inherits from the parent service's address. If the parent service's address is empty, then it will default to `127.0.0.1`. This mimics the previous logic, where if `sidecar.Proxy.LocalServiceAddress` was empty then it defaulted to `127.0.0.1` (https://github.com/hashicorp/consul/blob/main/agent/sidecar_service.go#L108-L110).